packages/components/bin/build-metadata.mts

```diff
@@ -0,0 +1,76 @@
+import { mkdir, writeFile } from 'node:fs/promises';
+import { join } from 'node:path';
+import { parse } from 'react-docgen-typescript';
+
+type ComponentMetadata = {
+	props: Record<
+		string,
+		{
+			description: string;
+			defaultValue?: { value: any };
+			required: boolean;
+			type: { name: string };
+		}
+	>;
+};
+
+const DOCUMENTED_COMPONENTS = [ 'breadcrumbs', 'summary-button' ];
+
+const files = DOCUMENTED_COMPONENTS.map( ( component ) => `${ component }/index.tsx` );
```

Comment:
I think in the future this would ideally be something we don't maintain as a list and instead `glob( 'src/*/index.tsx' )`. But `react-docgen-typescript` is pretty slow at parsing, and we have a lot of components -- most of which we don't plan on keeping here long-term. So for now I went with an allowlist.

Reply:
That's a good call for a more complex example. I incorporated a demonstration of support for subcomponents in d0d9da4401f . I don't know that we'll want to have a mapping like this long-term vs. something like a manifest file, and am open to feedback. I think this makes for a consolidated source of truth, though may not be especially scalable as we add more.